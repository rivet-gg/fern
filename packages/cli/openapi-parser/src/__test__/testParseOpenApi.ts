import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
