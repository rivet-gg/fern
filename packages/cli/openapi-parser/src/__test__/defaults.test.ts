import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("defaults", "openapi.yml");
});
