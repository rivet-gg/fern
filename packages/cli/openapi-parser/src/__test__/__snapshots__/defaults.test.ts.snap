// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser defaults parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [],
  "schemas": {
    "jointOwnerInvitationFields": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": "Fields used to compose other joint owner invitation schemas.",
      "generatedName": "JointOwnerInvitationFields",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "jointOwnerInvitationFieldsFirstName",
          "key": "firstName",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": "The invitee's first name.",
            "generatedName": "JointOwnerInvitationFieldsFirstName",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": "Jane",
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "jointOwnerInvitationFieldsLastName",
          "key": "lastName",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": "The invitee's last name.",
            "generatedName": "JointOwnerInvitationFieldsLastName",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": "Doe",
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "jointOwnerInvitationFieldsAge",
          "key": "age",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": "The invitee's age.",
            "generatedName": "JointOwnerInvitationFieldsAge",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": 1000000,
              "type": "int64",
            },
            "type": "primitive",
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "jointOwnerInvitationFieldsDisallowDebitCardAccess",
          "key": "disallowDebitCardAccess",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": "If \`true\`, the invitee is not allowed to order or manage debit cards for the account.",
            "generatedName": "JointOwnerInvitationFieldsDisallowDebitCardAccess",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": false,
              "type": "boolean",
            },
            "type": "primitive",
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {},
  "servers": [
    {
      "description": null,
      "name": null,
      "url": "https://api.apiture.com/banking",
    },
  ],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Copied mostly from Apiture",
  "variables": {},
  "webhooks": [],
}
`;
